flake.nix

```diff
@@ -0,0 +1,32 @@
+{
+  description = "Flake for Plutus";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs?rev=7d71001b796340b219d1bfa8552c81995017544a";
+    haskell-nix.url = "github:input-output-hk/haskell.nix?rev=87084d65a476cc826a0e8c5d281d494254f5bc7a";
+    flake-utils.url = "github:numtide/flake-utils?rev=b543720b25df6ffdfcf9227afafc5b8c1fabfae8";
+  };
+
+  outputs = { self, nixpkgs, haskell-nix, flake-utils, ... }:
+    (flake-utils.lib.eachSystem [ "x86_64-linux" ] (system:
+      let
+        topLevel = import ./. {
+          inherit system;
+          haskellNixOverlays = [ haskell-nix.overlay ];
```

Comment:
You're not passing haskell-nix here? That seems weird: surely then everything lower down would use the wrong haskell.nix. I guess that means it's unused, and we actually only care about the overlay: maybe we should delete the parameter.

Reply:
well here is what `default.nix` looks like:
```
{
...
, haskellNix ? import sources."haskell.nix" {
    sourcesOverride = {
      hackage = sources."hackage.nix";
      stackage = sources."stackage.nix";
    };
  }
, haskellNixOverlays ? haskellNix.overlays
...
}:
```

So if we really wanted to drop `haskellNix` we'd have to do some stringy business like defaulting to `"overlays"` and passing `"overlay"` from the flake.nix and then construct the `haskellNix."${haskellNixOverlay}"`. Better not I guess?

Reply:
`haskellNixOverlays = let haskellNix = ... in haskellNix.overlays`

The point is that having that `haskellNix` variable in scope inside `default.nix` is misleading: it can be incoherent with the overlay in the flake case, so we shouldn't have it.

Reply:
But inside `nix/default.nix` we are also accessing `haskellNix.config`:
```
  pkgs = import sources.nixpkgs {
    inherit system crossSystem;
    overlays = extraOverlays ++ overlays;
    config = haskellNix.config // config;
  };
```
So we'd have to pass that in as well (i haven't checked if the config contains anything relevant or likely it is to _become_ relevant if it isn't right now)

**Or** we could indeed also pass `haskellNix` to `default.nix` in the flake.

Reply:
I don't mind. If we do indeed pass `haskellNix` then maybe we're better off passing that alongside `isInFlake` or something so we don't need the overlay as well? it feels a little risky passing both of them, but :shrug:

Reply:
My latest commit has just that (passing in `haskellNix` and `isInFlake`) and i think that looks nice enough but the outputs of `nix-build marlowe-playground.client` and `nix build .#marlowe-playground-client` differ :sob: